src/cache/tlru-cache.ts

```diff
@@ -0,0 +1,77 @@
+import { LRUCache } from './lru-cache';
+
+/**
+ * Time-aware, least-recently-used cache (TLRU). Variant of LRU where entries have valid lifetime.
+ * @param {number} maxSize - Maximum cache size
+ * @param {number} ttl - Time in milliseconds after which cache entry will evict itself
+ * @param {number} evictionInterval - Frequency of cache entries eviction check
+ **/
+export class TLRUCache extends LRUCache {
+  private readonly cacheEntriesTTLRegistry = new Map<string, Date>();
+  constructor(readonly maxSize: number, readonly ttl: number) {
+    super(maxSize);
+  }
+
+  private getCacheEntryEvictionTime(): Date {
+    return new Date(Date.now() + this.ttl);
+  }
+
+  private clearCacheEntryEvictionTimeIfExists(key: string): void {
+    if (this.cacheEntriesTTLRegistry.has(key)) {
+      this.cacheEntriesTTLRegistry.delete(key);
+    }
+  }
+
+  private setCacheEntryEvictionTime(key: string): void {
+    this.cacheEntriesTTLRegistry.set(key, this.getCacheEntryEvictionTime());
+  }
+
+  private resetCacheEntryEvictionTime(key: string): void {
+    this.clearCacheEntryEvictionTimeIfExists(key);
+    this.setCacheEntryEvictionTime(key);
+  }
+
+  private evictExpiredCacheEntries() {
+    const now = new Date(Date.now());
+    let cacheKey: string;
+    let evictionDate: Date;
+
+    // Not using this.cache.forEach so we can break the loop once
+    // we find the fist non-expired entry. Each entry after that
+    // is guaranteed to also be non-expired, because they are oldest->newest
```

Comment:
Nice! "Iteration happens in insertion order" ([docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#description))

Geat to take advantage of this 💪